src/main/java/com/google/sps/servlets/MapServlet.java

```diff
@@ -0,0 +1,133 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.Trip;
+import com.google.sps.TripDay;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-map")
+public class MapServlet extends HttpServlet {
+
+  private final String TRIP_KEY_PARAM = "tripKey";
+
+  /**
+   * Checks for invalid cases (no user or tripKey).
+   * Gets the locations from datastore and prints them to writer.
+   */  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+    response.setContentType("application/json;");
+
+    // Initialize Datastore.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Get current user.
+    Entity userEntity = AuthServlet.getCurrentUserEntity();
+
+    // Get tripKey.
+    String stringTripKey = request.getParameter(TRIP_KEY_PARAM);
+
+    // if no tripKey return to trips page.
+    if (stringTripKey == null) {
+      response.getWriter().println("No trip Key");
+      response.sendRedirect("/trips/");
+  
+    // Redirects to sign in page if no user is signed in
+    // Otherwise redirects to create trips page
+    } else if (userEntity == null) {
+      response.getWriter().println("No current User");
+      response.sendRedirect("/");
+    } else { 
+      Key tripKey = KeyFactory.stringToKey(stringTripKey);
+
+      // Gets the locations from datastore and writes them to .../get-map
+      String result = doGetMap(response, datastore, userEntity, tripKey);
+      response.getWriter().println(result);
+    }
+  }
+
+  /**
+   * Gets the locations and prints them to writer.
```

Comment:
adjust the comment, as this method no longer prints

Reply:
Good catch, updated!